### ISSUE TITLE
U256 serialization is still not quantity

### ISSUE THREAD

**Issue by Rjected** [paradigmxyz/reth#1552]
### Describe the bug

While investigating why `ethereum/sync` was still showing the following error:
```
can't query chain head: json: cannot unmarshal hex number with leading zero digits into Go struct field Header.number of type *hexutil.Big
```
Actions run: https://github.com/paradigmxyz/reth/actions/runs/4267053919/jobs/7428426037

I ran the following:
```
reth import --chain ~/hive-data/genesis.json --path ~/hive-data/chain.rlp --db ~/hive-data
reth node --chain ~/hive-data/genesis.json --db ~/hive-data --disable-discovery --trusted-only --http
cast block latest
```

And this was the response:
```
dan@Dans-MacBook-Pro ~/p/reth (main) [1]> cast block latest
Error:
Deserialization Error: duplicate field `size` at line 1 column 1468. Response: {"hash":"0xc95596f4707fb382554b660b4847c599eb5f8fdcf99be2c5654aaadd4ec97840","parentHash":"0x08b2020a03028e5e786ff92c36b051be8235087e2e0554d38167b29be9f40bd8","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","author":"0x0000000000000000000000000000000000000000","miner":"0x0000000000000000000000000000000000000000","stateRoot":"0xe3f5a72cc1c86d794fd190473c775a774605059a45775c96e4df01205e40c007","transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","withdrawalsRoot":null,"number":"0x0bb8","gasUsed":"0x0","gasLimit":"0x80000000","extraData":"0x","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","timestamp":"0x080c9f88","difficulty":"0x020000","mixHash":"0x13beb8a6d5bb7540e846a92e4bc88627d76e7a1a61f55806c9e88cb4d81aa0ad","nonce":"0x340f24b19808a115","size":"0x01fd","totalDifficulty":"0x17720000","uncles":[],"transactions":[],"size":"0x0202","withdrawals":null}
```

There are two issues here:
 * There is in fact a duplicate `size` field
 * The `number` field is not quantity, because it has a leading zero:
 ```
 "number":"0x0bb8"
 ```
 It should instead be:
```
"number":"0xbb8"
```

### Steps to reproduce

Either running the hive `ethereum/sync` test suite, or doing the above procedure with a `chain.rlp` file and `genesis.json` should reproduce this error

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by gakonst** [comment]
Not only is the size field different, it also has two different values? 

Can you own investigating / fixing this?

**Comment by Rjected** [comment]
fixed in #1554
